tools/cldr/genclockprefs.js

```diff
@@ -38,189 +37,62 @@ function usage() {
+};
 
-function setLocaleData(data, locale) {
-    try {
-        if (locale.getScript()) {
-            if (locale.getRegion()) {
-                if (!localeData[locale.getLanguage()]) {
-                    localeData[locale.getLanguage()] = {};
-                }
-                if (!localeData[locale.getLanguage()][locale.getScript()]) {
-                    localeData[locale.getLanguage()][locale.getScript()] = {};
-                }
-                if (!localeData[locale.getLanguage()][locale.getScript()][locale.getRegion()]) {
-                    localeData[locale.getLanguage()][locale.getScript()][locale.getRegion()] = {};
-                }
-                localeData[locale.getLanguage()][locale.getScript()][locale.getRegion()].data = data;
-            }
-        } else if (locale.getRegion()) {
-            if (!localeData[locale.getLanguage()]) {
-                localeData[locale.getLanguage()] = {};
-            }
-            if (!localeData[locale.getLanguage()][locale.getRegion()]) {
-                localeData[locale.getLanguage()][locale.getRegion()] = {};
-            }
-            localeData[locale.getLanguage()][locale.getRegion()].data = data;
-        } else if (locale.getLanguage()) {
-            if (!localeData[locale.getLanguage()]) {
-                localeData[locale.getLanguage()] = {};
-            }
-            localeData[locale.getLanguage()].data = data;
-        } else {
-            // root locale#
-            localeData.data = data;
-        }
-    } catch (e) {
-        return undefined;
-    }
-    return data;
-}
-
-function anyProperties(data) {
-    var count = 0;
-    for (var prop in data) {
-        if (prop && data[prop]) {
-            count++;
-        }
-        if (count > 1) {
-            return true;
-        }
-    }
-    return false;
-}
+clockprefs["clock"] = rootClockPref;
+fs.writeFileSync(path.join(resultDir, "clock.jf"), JSON.stringify(clockprefs, true, 4), "utf-8");
 
-function writeClockPrefs(language, script, region, data) {
-    var pathname = calcLocalePath(language, script, region, "");
-    //console.log("data to be written into jf files" + pathname + "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"+JSON.stringify(data));
-    if (data.generated) {
-        if (anyProperties(data)) {
-            console.log("Writing " + pathname);
-            makeDirs(pathname);
-            //console.log("data to be written into jf files" + pathname + "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"+JSON.stringifyta["clock"]));#
-            if (typeof (data["clock"]) != 'undefined') {
-                data.generated = true;
-                makeDirs(pathname);
-                fs.writeFileSync(path.join(pathname, "clock.jf"), JSON.stringify(data, true, 4), "utf-8");
-            }
-        } else {
-            console.log("Skipping empty " + pathname);
-        }
-    } else {
-        console.log("Skipping existing " + pathname);
-    }
-}
-
-function getClockPrefs(locale) {
+for(var region in timeData){
```

Comment:
This is a locale, not a region. It took me a little time to realize that. Maybe rename the variable to "locale" ? Also maybe use `new Locale(locale)` to parse it instead of substrings?

Reply:
oh, I will rename the value as locale.  Most the data  in timeData.json is represent region so I thought not using `new Locale` would become code simpler  and doesn't need to have ilib dependency. If you think it's better to use new Locale for parsing locale, I'll update it too.

Reply:
There is common.Locale imported at the top of the file, so it doesn't have to rely on ilib itself

Reply:
you are right. I updated the locale parsing part.